cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -69,12 +68,6 @@ func PerformPostUpgradeTasks(client clientset.Interface, cfg *kubeadmapi.InitCon
 		errs = append(errs, err)
 	}
 
-	// TODO: Temporary workaround. Remove in 1.25:
-	// https://github.com/kubernetes/kubeadm/issues/2426
-	if err := UpdateKubeletDynamicEnvFileWithURLScheme(dryRun); err != nil {
-		return err
-	}
-
```

Comment:
note for the future.

we might want to use a similar function for other mutations in the dynamic env file.
e.g. when we have to do some cleanups of the deprecated kubelet flags such as --container-runtime=remote
https://github.com/kubernetes/kubeadm/issues/2626
https://github.com/kubernetes/kubernetes/issues/106893